os/board/rtl8721csm/src/component/os/tizenrt/rtk_netmgr.c

```diff
@@ -394,16 +394,31 @@ trwifi_result_e wifi_netmgr_utils_deinit(struct netdev *dev)
 trwifi_result_e wifi_netmgr_utils_scan_ap(struct netdev *dev, trwifi_scan_config_s *config)
 {
 	if (config) {
-		if (config->ssid_length > 0) {
+		 if ((config->channel >= 1) && (config->channel <= 13)) {
+			uint32_t	channel;
+			uint8_t     pscan_config;
+			if (config->ssid_length == 0) {
+				rtw_memset(config->ssid, 0, sizeof(config->ssid));
+				config->ssid_length = 0;
+			}
+			channel = config->channel;
+			channel &= 0xff;
+			pscan_config = PSCAN_ENABLE;
+			if (wifi_set_pscan_chan((uint8_t *)&channel, &pscan_config, 1) < 0 ) {
+				RTW_API_INFO("set pscan channel failed\n");
+				return TRWIFI_FAIL;
+			}
+		}
+		if ((config->ssid_length > 0) || ((config->channel >= 1) && (config->channel <= 13))) {

```

Comment:
@pillip8282, we are checking only if it is full scan "wm_test scan" or partial scan which covers 3 other cases "wm_test scan ssid [ssid]" / "wm_test scan ch [channel]" / "wm_test scan both [ssid] [channel]" using `if (config)`

The full scan uses `wifi_scan_networks` API. All other three cases uses the same API `wifi_scan_networks_with_ssid` and the driver does case specific operations based on `trwifi_scan_config_s *config`.

Out of three partial scan cases, for two cases with `config->channel` we need to set pscan `wifi_set_pscan_chan` before `wifi_scan_networks_with_ssid`,
1) wm_test scan ch [channel]
2) wm_test scan both [ssid] [channel]

Reply:
Thanks for explanation. I understand it now. Because following code started to if, So after wifi_set_pscan_chan() wifi_scan_networks_with_ssid() can be called. 
```
if (( config->ssid_length > 0) || ((config->channel >= 1) && (config->channel <= 13))) {
```